### PR TITLE
Process Improvement: v0.18.0

### PR DESCRIPTION
**Pull request by galargh**
Related to https://github.com/ipfs/kubo/issues/9417

This PR is to track the process improvement as identified during the v0.18.0 release.

The new version of the release issue template available at https://github.com/ipfs/kubo/blob/process-improvement-v0.18.0/docs/RELEASE_ISSUE_TEMPLATE.md

### CODE REVIEW

**Comment by BigLep** [comment]
For the Bifrost gateway deployment, it's probably worth adding a pointer to Bifrost's docs: https://www.notion.so/pl-strflt/Kubo-Gateway-Release-Notes-6e0efff28ee540be9ccb8f2b85104c42

**Comment by BigLep** [comment]
I think we should add a post-release step for creating the PR that updates our dependencies.  As long as the PR gets created and linked to the the next release, we are good.  Something like:

```
- [ ] Initiate dependence update
   - [ ] Run X to update all dependencies and create PR (example)
   - [ ] Link new PR to next release issue that was created above
```

**Comment by BigLep** [comment]
I noticed that https://github.com/ipfs/kubo/blob/master/docs/changelogs/v0.18.md doesn't include contributors or changelog filled in.

**Comment by BigLep** [comment]
@galargh : doh - my bad.  I think I must have been looking at the file in a branch before #9581 was merged!  Thanks!

File: docs/RELEASE_ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -38,176 +37,190 @@ As usual, this release includes important fixes, some of which may be critical f
...
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) Create the next [changelog](https://github.com/ipfs/kubo/blob/master/docs/changelogs/vX.(Y+1).md)
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) Link to the new changelog in the [CHANGELOG.md](CHANGELOG.md) file
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) Create the next release issue
+  </details>
+- [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) Create a dependency update PR
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) check out [ipfs/kubo](https://github.com/ipfs/kubo)
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) run `go get -t -u ./...`
```
Comment (galargh) [inline docs/RELEASE_ISSUE_TEMPLATE.md:220]:
```suggestion
  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) run `go get -t -u`
```

E.g. https://github.com/ipfs/kubo/pull/9618

Comment (galargh) [inline docs/RELEASE_ISSUE_TEMPLATE.md:220]:
```suggestion
  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) run `go get -u` in root directory
  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) run `go mod tidy` in root directory
  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) ![](https://img.shields.io/badge/not-PATCH-yellow?style=flat-square) run `go mod tidy` in `docs/examples/kubo-as-a-library` directory
```
Tested in: https://github.com/ipfs/kubo/pull/9619

**Comment by galargh** [comment]
The release instructions tested during v0.18.1: https://github.com/ipfs/kubo/blob/process-improvement-v0.18.0/docs/RELEASE_ISSUE_TEMPLATE.md

File: docs/RELEASE_ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -38,176 +39,193 @@ As usual, this release includes important fixes, some of which may be critical f
...
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) check out [ipfs/interop](https://github.com/ipfs/interop)
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) run `npm install`
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) create a PR which updates `package.json` and `package-lock.json`
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) merge the PR
+  </details>
+- [ ] Update Kubo in [ipfs-desktop](https://github.com/ipfs/ipfs-desktop) <details><summary>using `kuboreleaser release --version vX.Y.Z(-RCN) update-ipfs-desktop` or ...</summary>
+  - [ ] check out [ipfs/ipfs-desktop](https://github.com/ipfs/ipfs-desktop)
+  - [ ] run `npm install`
+  - [ ] create a PR which updates `package.json` and `package-lock.json`
+  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) merge the PR
```
Comment (SgtPooki) [inline docs/RELEASE_ISSUE_TEMPLATE.md:197]: Can we add a step here to ensure that @SgtPooki or @whizzzkid are tagged on this PR that is created? Also, can we add a note that unless you're performing the ipfs-desktop release, you should not merge that PR unless you want the ipfs-desktop maintainers to miss that we need to do a release :)

see https://github.com/ipfs/ipfs-desktop/issues/2407#issuecomment-1437430617

Comment (galargh) [reply docs/RELEASE_ISSUE_TEMPLATE.md:197]:
```suggestion
- [ ] Update Kubo in [ipfs-desktop](https://github.com/ipfs/ipfs-desktop) <details><summary>using `kuboreleaser release --version vX.Y.Z(-RCN) update-ipfs-desktop` or ...</summary>
  - [ ] check out [ipfs/ipfs-desktop](https://github.com/ipfs/ipfs-desktop)
  - [ ] run `npm install`
  - [ ] create a PR which updates `package.json` and `package-lock.json`
  - [ ] ![](https://img.shields.io/badge/only-FINAL-green?style=flat-square) add @SgtPooki and @whizzzkid as reviewers
```

**Comment by galargh** [comment]
@BigLep We tested these instructions during v0.19.0. Let me know if you're OK with merging this now.

**Pull request merged**